### ISSUE TITLE
[request] uconstr_eq

### ISSUE THREAD

**Issue by JasonGross** [coq/coq#6546]
#### Version

master

#### Description of the problem

Sometimes, as in #6545, the cost of `constr_eq` is dominated by the time it takes to build the `constr` argument.  I'd like a variant `uconstr_eq` that accepts uconstrs.  I think `uconstr_eq _ _` should succeed, and `uconstr_eq _ <any term which is not _>` should fail (that is, holes should compare equal with holes and compare not equal with terms).

**Comment by ppedrot** [comment]
I don't think you can properly test uconstrs for equality, you have to evaluate them first to get a normal form. That's the whole point about uconstrs.
